Documentation/Open Location Code API Specification.md

```diff
@@ -0,0 +1,78 @@
+# Open Location Code API Reference Specification
+
+## Version history
+
+All substantial changes changes to this document are listed here.
+
+* Version 1.1.0 / 2021-06-04 / [William Entriken](https://github.com/fulldecent/)
+  * Update to match nomenclature in [Plus Codes specification](./Plus%20Codes%20Specification.md)
+  * Separate REQUIRED and OPTIONAL public methods
+  * Specify the number of significant digits that a short code Plus Code may omit
+
+- Version 1.0.0 / 2014-10-27 / [Doug Rinckes](https://github.com/drinckes), Google / Philipp Bunge, Google
+  - Initial public release
+
+## REQUIRED public methods
+
+An implementation of the Open Location Code API SHALL implement these REQUIRED methods:
+
+| Method name          | Input(s)                        | Output(s)                                                    |
+| -------------------- | ------------------------------- | ------------------------------------------------------------ |
+| `isFull`             | Plus Code string                | True if this is a long code Plus Code, false otherwise       |
+| `encode`             | Latitude, longitude             | The Plus Code with code length 10 with area that contains the specified coordinates |
+| `decode`             | Full code Plus Code string with length 10 | The southern parallel and western meridian for the Plus Code area |
+
+Implementations are RECOMMENDED to use the method names above.
+
+Note: it is possible to implement `isFull` using a Perl Compatible Regular Expression, see the Plus Code specification.
+
+## OPTIONAL public methods
```

Comment:
Current API.txt does not seem to consider exactly the above three methods as REQUIRED and all remaining ones as OPTIONAL. I see the point of not forcing implementations to offer "shorten"-related functionality, but I think that at least `isValid` and perhaps `encode/decode` with variable code length might as well be required. Perhaps check what the current implementations do, then let's discuss?

Reply:
Also I moved `isValid`, which regards full+short codes from required to optional. This is consistent with all other things relating to short codes being optional. Separately, no implementation implements `isValid` currently as specified (see below).

Breaking change warning added.

🦶 : Progress at ce6bb6db03524857f7b3762b5c6d3236b02428b3 // marking below as an open item to review implementations and which methods should be required 

Replaced specifications:

* https://github.com/google/open-location-code/blob/main/API.txt
* https://github.com/google/open-location-code/blob/a323e80159d6d6170158cc22497250cb86bf3583/docs/specification.md#library-implementation-requirements

Reply:
Whether or not implementations  currently implement a required method correctly shouldn't play a role in determining whether or not that method should stay required.

Also, I think that making `isValid` optional without any kind of common replacement is kind of backwards. A valid code (full or short) is a valid code, whether or not any given implementation is able to handle it.

One hypothetical replacement might be `canBeDecoded`, a boolean that would return true iff the code is valid and full, OR if it is valid, short and the implementation in question implements the optional bits. However, this definition already uses a code's validity again, so the `isValid` method might as well keep being required.

Reply:
Fixed `isValid` at 362c8ecd975cebd037233f4f6411eea94818e134